### DISCUSSION TITLE
How do I compare a datetime created by Pydantic with one created by python's datetime.fromisoformat()?

### DISCUSSION THREAD

**Discussion by ratkins** [pydantic/pydantic#8300] [Question]
Apparently, when I annotate a `datetime` field in a `BaseModel` subclass and Pydantic creates the object for me, the runtime type of the datetime field is the (private) Pydantic type `TzInfo`.

Unfortunately, this doesn't compare as equal to the object I get back from creating one with `datetime.timezone(timedelta(hours=1))`. Which means my `.assert_called_with()` assertion fails in my test, even when the objects legitimately represent the same instant:

```
Expected: [...] event_timestamp=datetime.datetime(2023, 12, 4, 11, 15, 58, tzinfo=datetime.timezone(datetime.timedelta(seconds=3600))), timestamp_created=None))
Actual: [...].  event_timestamp=datetime.datetime(2023, 12, 4, 11, 15, 58, tzinfo=TzInfo(+01:00)), timestamp_created=None))
```

Can I somehow instruct pydantic to construct a canonicalised `datetime` object I can compare with a `datetime` object it's possible to construct myself?

**Comment by PrimozGodec** [comment]
@ratkins, did you find any solution?

**Reply by ybressler** [reply]
Can you share an excerpt for replicating your result? I will look into this once you do.

**Reply by ybressler** [reply]
Also, possibly this could be helpful? [Specify timezone in pydantic datetime field [Python]](https://stackoverflow.com/questions/73441112/specify-timezone-in-pydantic-datetime-field-python)
